photon_stream/lowlevel.py

```diff
@@ -57,15 +63,46 @@ def flatten(self, start_time=15e-9, end_time=65e-9):
         for px, pixel_photons in enumerate(self.time_lines):
             for photon_slice in pixel_photons:
                     xyt.append([
-                        geo.pixel_azimuth[px],
-                        geo.pixel_zenith[px],
+                        pixels.azimuth.iloc[px],
+                        pixels.zenith.iloc[px],
                         photon_slice * self.slice_duration
                         ])
         xyt = np.array(xyt)
         past_start = xyt[:, 2] >= start_time
         before_end = xyt[:, 2] <= end_time
         return xyt[past_start*before_end]
 
+    @property
+    def labels(self):
+        if not hasattr(self, '_labels'):
+            self._cluster()
+        return self._labels
+
+    @property
+    def number_of_clusters(self):
+        if not hasattr(self, '_number_of_clusters'):
+            self._cluster()
+        return self._number_of_clusters
+
+    def _cluster(self, eps=0.1, min_samples=20, deg_over_s=0.35e9):
```

Comment:
I fear the photon stream gets too much responsibility here. Clusters should be a stand alone thing, so that there can be e.g. several different clusters, with different thresholds and settings for the dame photon stream